ray-operator/controllers/ray/utils/util.go

```diff
@@ -71,6 +71,40 @@ func IsCreated(pod *corev1.Pod) bool {
 	return pod.Status.Phase != ""
 }
 
+func FindHeadPodReadyCondition(pods corev1.PodList) metav1.Condition {
+	var headPod *corev1.Pod
```

Comment:
How about change to `func FindPodReadyCondition(pod corev1.Pod) metav1.Condition` instead? Then, we pass head Pod into the function (we can wait for [getHeadPod](https://github.com/ray-project/kuberay/pull/2266#discussion_r1689064856)). We may have thousands of Pods in a RayCluster. The loop may be a bit inefficient.

Reply:
The getHeadPod will be moved to `common.GetRayClusterHeadPod` in the waiting PR.

Reply:
merge @rueian 's PR and use it here
https://github.com/ray-project/kuberay/blob/41bd563f3af567efed34bd3181957f1005e842eb/ray-operator/controllers/ray/raycluster_controller.go#L1208-L1212